test/verify/check-machines-dbus

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import parent
+import unittest
+
+from testlib import *
+import machineslib
+
+
+class TestMachinesDBus(machineslib.TestMachines):
+
+    def detectLibvirtDbus(self):
+        m = self.machine
+        namesList = m.execute("dbus-send --system --dest=org.freedesktop.DBus "
+                              "--type=method_call "
+                              "--print-reply /org/freedesktop/DBus "
+                              "org.freedesktop.DBus.ListActivatableNames")
+
+        if ('org.libvirt' not in namesList):
+            raise unittest.SkipTest("libvirt-dbus is not available");
+        m.execute("echo 'PROVIDER=libvirt-dbus' > /etc/environment")
```

Comment:
I think this is just a flag for machineslib.py, isn't it? It would IMHO be better to pass that as a class property in `class TestMachinesDBus`, and set `provider = "dbus"` here, and "virsh" in the other one, and then machineslib.py can just query that. This doesn't need to get all the way into the testbed.

Reply:
TestMachinesDBus is the child class, and TestMachines is the parent class and we are inheriting the whole test suite from TestMachines.
Is it possible to query attributes of the child class from the parent class?

Reply:
I just realized I can set the attribute provider to the parent class and just override it.
Done